src/ui/win32/Desktop.cpp

```diff
@@ -12,43 +13,69 @@ namespace win32 {
 
 Desktop::Desktop() noexcept
 {
-    m_vDialogPresenters.emplace_back(new Dlg_MessageBox::Presenter());
+    m_vDialogPresenters.emplace_back(new MessageBoxDialog::Presenter());
+    m_vDialogPresenters.emplace_back(new RichPresenceDialog::Presenter());
 }
 
 void Desktop::ShowWindow(WindowViewModelBase& vmViewModel) const
 {
-    auto* pController = GetDialogController(vmViewModel);
-    if (pController != nullptr)
+    auto* pPresenter = GetDialogPresenter(vmViewModel);
+    if (pPresenter != nullptr)
     {
-        pController->ShowWindow(vmViewModel);
+        pPresenter->ShowWindow(vmViewModel);
     }
     else
     {
-        assert("!No controller for view model");
+        assert("!No presenter for view model");
     }
 }
 
 ra::ui::DialogResult Desktop::ShowModal(WindowViewModelBase& vmViewModel) const
 {
-    auto* pController = GetDialogController(vmViewModel);
-    if (pController != nullptr)
+    auto* pPresenter = GetDialogPresenter(vmViewModel);
+    if (pPresenter != nullptr)
     {
-        pController->ShowModal(vmViewModel);
+        pPresenter->ShowModal(vmViewModel);
     }
     else
     {
-        assert("!No controller for view model");
+        assert("!No presenter for view model");
     }
 
     return vmViewModel.GetDialogResult();
 }
 
-ra::ui::win32::IDialogPresenter* Desktop::GetDialogController(ra::ui::WindowViewModelBase& oViewModel) const
+void Desktop::GetWorkArea(ra::ui::Position& oUpperLeftCorner, ra::ui::Size& oSize) const
 {
-    for (auto& pController : m_vDialogPresenters)
+    if (m_oWorkAreaSize.Width == 0)
     {
-        if (pController->IsSupported(oViewModel))
-            return pController.get();
+        RECT rcWorkArea;
+        if (SystemParametersInfo(SPI_GETWORKAREA, 0, &rcWorkArea, 0))
+        {
+            m_oWorkAreaPosition = { rcWorkArea.left, rcWorkArea.top };
+            m_oWorkAreaSize = { rcWorkArea.right - rcWorkArea.left, rcWorkArea.bottom - rcWorkArea.top };
+
+            RA_LOG_INFO("Desktop work area: %dx%d", m_oWorkAreaSize.Width, m_oWorkAreaSize.Height);
+        }
+        else
+        {
+            RA_LOG_ERR("Unable to determine work area");
+
+            m_oWorkAreaPosition = { 0, 0 };
+            m_oWorkAreaSize = { 800, 600 };
+        }
+    }
+
+    oUpperLeftCorner = m_oWorkAreaPosition;
+    oSize = m_oWorkAreaSize;
+}
+
+ra::ui::win32::IDialogPresenter* Desktop::GetDialogPresenter(ra::ui::WindowViewModelBase& oViewModel) const
```

Comment:
This right here looks way too verbose (the `ra::ui::win32::IDialogPresenter*` part). I'd probably pull my hair out if I had to the write the fully qualified name each time when it could've been avoided. It's too bad Visual Studio only has built-in ReSharper-like capabilities for .NET Framework languages.